### ISSUE TITLE
Remote - SSH connection issue

### ISSUE THREAD

**Issue by gaojinhsu** [microsoft/vscode#221336]
- VS Code Version: 
```
            Version: 1.91.0 (Universal)
            Commit: ea1445cc7016315d0f5728f8e8b12a45dc0a7286
            Date: 2024-07-01T18:54:18.541Z
            Electron: 29.4.0
            ElectronBuildId: 9728852
            Chromium: 122.0.6261.156
            Node.js: 20.9.0
            V8: 12.2.281.27-electron.0
            OS: Darwin arm64 23.5.0
```
- OS Version: macOS 14.5
- Remote machine OS Version： macOS 11.6.1
- vscode server log: 
``` Visual Studio Code Server

 By using the software, you agree to
 the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
 the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).

Server bound to /var/folders/pc/22d4wg1d1xnd1mpw4nd8k9hh0000gn/T/code-50632d34-2d39-4d32-bffe-803ff818098d
Extension host agent listening on /var/folders/pc/22d4wg1d1xnd1mpw4nd8k9hh0000gn/T/code-50632d34-2d39-4d32-bffe-803ff818098d

FATAL ERROR: v8::Template::Set Invalid value, must be a primitive or a Template
----- Native stack trace -----

 1: 0x1028766c5 node::Abort() [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
 2: 0x1028767b1 node::OnFatalError(char const*, char const*) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
 3: 0x102a2b055 v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
 4: 0x1028231a7 napi_define_class [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
 5: 0x108693a38 Napi::ObjectWrap<Logger>::DefineClass(Napi::Env, char const*, unsigned long, napi_property_descriptor const*, void*) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node_modules/@vscode/spdlog/build/Release/spdlog.node]
 6: 0x10868d709 Logger::Init(Napi::Env, Napi::Object) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node_modules/@vscode/spdlog/build/Release/spdlog.node]
 7: 0x108662388 Init(Napi::Env, Napi::Object) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node_modules/@vscode/spdlog/build/Release/spdlog.node]
 8: 0x108662551 napi_register_module_v1 [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node_modules/@vscode/spdlog/build/Release/spdlog.node]
 9: 0x10283d72d napi_module_register_by_symbol(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, napi_value__* (*)(napi_env__*, napi_value__*), int) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
10: 0x102843819 std::__1::__function::__func<node::binding::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)::$_0, std::__1::allocator<node::binding::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)::$_0>, bool (node::binding::DLib*)>::operator()(node::binding::DLib*&&) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
11: 0x10280d323 node::Environment::TryLoadAddon(char const*, int, std::__1::function<bool (node::binding::DLib*)> const&) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
12: 0x1028420cf node::binding::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
13: 0x102a90da8 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, unsigned long*, int) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
14: 0x102a9036a v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
15: 0x10343d1b6 Builtins_CEntry_Return1_ArgvOnStack_BuiltinExit [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
16: 0x1033aec9c Builtins_InterpreterEntryTrampoline [/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node]
17: 0x110f73e92 
18: 0x110f5d50f 

----- JavaScript stack trace -----

1: Module._extensions..node (node:internal/modules/cjs/loader:1473:18)
2: Module.load (node:internal/modules/cjs/loader:1207:32)
3: Module._load (node:internal/modules/cjs/loader:1023:12)
4: Module.require (node:internal/modules/cjs/loader:1235:19)
5: require (node:internal/modules/helpers:176:18)
6: bindings (/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node_modules/bindings/bindings.js:112:48)
7: /Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/node_modules/@vscode/spdlog/index.js:3:35
8: Module._compile (node:internal/modules/cjs/loader:1376:14)
9: Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
10: Module.load (node:internal/modules/cjs/loader:1207:32)


/Users/bkdevops/.vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/bin/code-server: line 22:  8032 Abort trap: 6           "$ROOT/node" ${INSPECT:-} "$ROOT/out/server-main.js" "$@"
```
- local error tips:
<img width="493" alt="image" src="https://github.com/microsoft/vscode/assets/2178687/f5d03f75-9c78-41cc-b913-5912e568de4f">

Steps to Reproduce:

1.just use remote-ssh (v0.112.0) to connect a remote mac machine

**Comment by gaojinhsu** [comment]
it works well on this version
```
Version: 1.90.2
Commit: 5437499feb04f7a586f677b155b039bc2b3669eb
Date: 2024-06-18T22:54:35.075Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0
```

**Comment by deepak1556** [comment]
Should be addressed in `1.91.1`

/duplicate https://github.com/microsoft/vscode/issues/219893
